File: core_py/examples/add_sheet.py
import time

from report_core import XLSXBook

if __name__ == "__main__":
    start_time = time.time()

    book = XLSXBook()

    sheet = book.add_sheet("A", 100, 100)
    print("Sheet", sheet)

    print("Write Cell 150x150")
    cell = sheet.write_cell(150, 150, "Жопа")

    sheet.generate_empty_cells()

    cell.set_formula("SUM(A1:A10)")
    cell.set_style_id("Style: Percent")

    print("Write Cell 2x2")
    cell = sheet.write_cell(2, 2, "100")

    print(
        "Find Cell 2x2 value:",
        (cell.value, type(cell.value)) if cell else cell,
    )

    print("Write Cell 3x3")
    cell = sheet.write_cell(3, 3, "300.0")

    print(
        "Find Cell 3x3 value:",
        (cell.value, type(cell.value)) if cell else cell,
    )

    for cell in sheet.cells:
        if cell.row == 99 and cell.column == 99:
            cell.set_value("Yop! Жопа")
            cell.set_style_id("Style Yop! Жопа")

        if cell.row == 100 and cell.column == 100:
            cell.set_formula("=A1")

    cell = sheet.find_cell_by_coords(100, 100)
    print(
        "Find Cell 100x100 value:",
        cell.value if cell else cell,
        "formula:",
        cell.formula if cell else cell,
        "StyleID:",
        cell.style_id if cell else cell,
        "IF formula:",
        cell.is_formula() if cell else cell,
        "Data type:",
        cell.data_type if cell else cell,
    )

    cell = sheet.find_cell_by_coords(99, 99)
    print(
        "Find Cell 99x99 value:",
        cell.value if cell else cell,
        "StyleID:",
        cell.style_id if cell else cell,
    )

    cell = sheet.find_cell_by_coords(120, 120)
    print("Find Cell 120x120", cell)

    print(f"Выполнено за: {time.time() - start_time:.3f} сек.")

File: core_rs/examples/add_sheet.rs
use std::time::Instant;

use core_rs::writer::book::XLSXBook;

fn main() -> anyhow::Result<()> {
    let start = Instant::now();

    let book = XLSXBook::new();

    let sheet = book.lock().add_sheet("A".to_string(), Some(50), Some(30));

    for cell in sheet.lock().cells() {
        let mut cell = cell.lock();

        if cell.row == 2 && cell.column == 2 {
            cell.set_value("100".to_string())?;
        }

        if cell.row == 3 && cell.column == 3 {
            cell.set_value("100.0".to_string())?;
        }

        if cell.row == 5 && cell.column == 5 {
            cell.set_value("AAAAA".to_string())?;
            cell.set_formula("=SUM:A10".to_string())?;
        }
    }

    let cell = sheet.lock().find_cell_by_coords(2, 2)?;
    println!("Find Coords {:?}", cell);

    let cell = sheet.lock().find_cell_by_coords(3, 3)?;
    println!("Find Coords {:?}", cell);

    let cell = sheet.lock().find_cell_by_coords(5, 5)?;
    println!("Find Coords {:?}", cell);

    let end = start.elapsed();
    println!(
        "Выполнено за: {}.{:03} сек.",
        end.as_secs(),
        end.subsec_millis(),
    );

    let start = Instant::now();
    let sheet = book.lock().add_sheet("B".to_string(), Some(50), Some(30));

    for r in 1..=1000 {
        for c in 1..=1000 {
            let cell = sheet.lock().write_cell(r, c, &format!("Yop! {}{}", r, c))?;

            let mut guarg_cell = cell.lock();
            if guarg_cell.row == 20 && guarg_cell.column == 20 {
                guarg_cell.set_value("AAAAA".to_string())?;
                guarg_cell.set_formula("=SUM(A1:A10)".to_string())?;
            }
        }
    }

    sheet.lock().write_cell(120, 120, "Yop! 120x120")?;

    sheet.lock().generate_empty_cells()?;

    println!(
        "Sheet len cells {:?}",
        sheet.lock().cells().collect::<Vec<_>>().len()
    );

    let cell = sheet.lock().find_cell_by_coords(1, 1)?;
    println!("Find Coords 1x1 {:?}", cell);

    let cell = sheet.lock().find_cell_by_coords(20, 20)?;
    println!("Find Coords 20x20 {:?}", cell);

    let cell = sheet.lock().find_cell_by_coords(105, 105)?;
    println!("Find Coords 105x105 {:?}", cell);

    let end = start.elapsed();
    println!(
        "Выполнено за: {}.{:03} сек.",
        end.as_secs(),
        end.subsec_millis(),
    );

    Ok(())
}
